### DISCUSSION TITLE
Currently avialable Script Scopes

### DISCUSSION THREAD

**Discussion by shaqtsui** [xmake-io/xmake#3482] [Q&A] [answered]
Base on document: `Currently available configuration scopes are: target(), option(), task(), package()`.
What about: `rule` & `toolchain`? 
Is this a document error? Or they are different from sub scope?

**Comment by waruqi** [comment, marked as answer]
They were added later and are now available. 

https://xmake.io/#/manual/custom_rule
https://xmake.io/#/manual/custom_toolchain

**Reply by shaqtsui** [reply]
Seems document need to be updated. I hope I can help.
